pkgs/btcpayserver/default.nix

```diff
@@ -0,0 +1,75 @@
+{ stdenv, fetchFromGitHub, fetchurl, nbxplorer, dotnetPackages, dotnetCorePackages, lib }:
+
+let
+
+  deps = import ./deps.nix {inherit fetchurl;};
+  dotnet-sdk = dotnetCorePackages.sdk_3_1;
+
+in
+
+stdenv.mkDerivation rec {
+  name = "btcpayserver";
+  version = "1.0.5.4";
+
+  src = fetchFromGitHub {
+    owner = "btcpayserver";
+    repo = "btcpayserver";
+    rev = "v${version}";
+    sha256 = "1h4567di6sfr6z734b7a7qh0hv5ywq14hlbimxcfxn9hri2g3zbv";
+  };
+  buildInputs = [dotnet-sdk dotnetPackages.Nuget];
+
+  buildPhase = ''
+    mkdir home
+    export HOME=$PWD/home
+
+    export DOTNET_CLI_TELEMETRY_OPTOUT=1
+    export DOTNET_SKIP_FIRST_EXPERIENCE=1
+
+    for package in ${toString deps}; do
+      nuget add $package -Source nixos
+    done
+
+    dotnet restore --source nixos BTCPayServer.Client/BTCPayServer.Client.csproj
+    dotnet build --no-restore -c Release BTCPayServer.Client/BTCPayServer.Client.csproj
+
+    dotnet restore --source nixos BTCPayServer.Data/BTCPayServer.Data.csproj
+    dotnet build --no-restore -c Release BTCPayServer.Data/BTCPayServer.Data.csproj
+
+    dotnet restore --source nixos BTCPayServer.Common/BTCPayServer.Common.csproj
+    dotnet build --no-restore -c Release BTCPayServer.Common/BTCPayServer.Common.csproj
+
+    dotnet restore --source nixos BTCPayServer.Rating/BTCPayServer.Rating.csproj
+    dotnet build --no-restore -c Release BTCPayServer.Rating/BTCPayServer.Rating.csproj
+
+    dotnet restore --source nixos BTCPayServer/BTCPayServer.csproj
+    dotnet build --no-restore -c Release BTCPayServer/BTCPayServer.csproj
+  '';
+
+  installPhase = ''
+      mkdir -p $out/bin/
+      echo -e '#!/usr/bin/env bash \n#launch BTCpayserver \nscriptdir=`dirname $(readlink $(which btcpayserver))` \ndotnet run --no-launch-profile --no-build -c Release -p "$scriptdir/../BTCPayServer/BTCPayServer.csproj" -- $@
```

Comment:
`dotnet` needs nix-store path substitution. I achieve this by using sed: 

https://github.com/fort-nix/nix-bitcoin/blob/33c17b2e44c1bb889f4cead682c77de454aabef4/pkgs/btcpayserver/dotnet-build.nix#L118

https://github.com/fort-nix/nix-bitcoin/blob/33c17b2e44c1bb889f4cead682c77de454aabef4/pkgs/btcpayserver/dotnet-build.nix#L167

Reply:
I solved this problem by doing

```diff
-      echo -e '#!/usr/bin/env bash \n#launch BTCpayserver \nscriptdir=`dirname $(readlink $(which btcpayserver))` \ndotnet run --no-launch-profile --no-build -c Release -p "$scriptdir/../BTCPayServer/BTCPayServer.csproj" -- $@
-' > $out/bin/run.sh
+      echo -e "#!/usr/bin/env bash \n#launch BTCpayserver \ndotnet run --no-launch-profile --no-build -c Release -p \"$out/BTCPayServer/BTCPayServer.csproj\" -- \$@" > $out/bin/run.sh
```
and
```diff
-    echo -e '#!/usr/bin/env bash \n#launch NBXplorer \nscriptdir=`dirname $(readlink $(which NBXplorer))` \ndotnet exec --additionalprobingpath $scriptdir/../home/.nuget/packages/ $scriptdir/../NBXplorer/bin/Release/netcoreapp3.1/NBXplorer.dll $@' > $out/bin/run.sh
+    echo -e "#!/usr/bin/env bash \n#launch NBXplorer \ndotnet exec --additionalprobingpath $out/home/.nuget/packages/ $out/NBXplorer/bin/Release/netcoreapp3.1/NBXplorer.dll \$@" > $out/bin/run.sh
```

Reply:
Fixed the impurities by directly using ```$out```

Reply:
Improved upon that with the method @jb55 suggested